CalendarApp/Change Handlers/UpdateBuilder.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  ChangeBuilder.swift
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 7/29/22.
+//
+
+import Foundation
+import CoreData
+
+@objc protocol CreateRevisionDelegate {
+    func createUpdateRevision(changeField: ChangeField) -> Revision
+}
+
+@objcMembers
+class UpdateBuilder : NSObject {
+    private var oldEvent: Event
+    private var newEvent: Event
+    private var delegate: CreateRevisionDelegate
+    
+    public required init(firstEvent: Event,
+                         updatedEvent: Event,
+                         inputDelegate: CreateRevisionDelegate) {
+        oldEvent = firstEvent
+        newEvent = updatedEvent
+        delegate = inputDelegate
+    }
+    
+    public func checkFields() {
```

Comment:
This should just be a method that takes events and spits out a revision. There is a lot of stateful stuff floating around this PR that doesn't need to be stateful.